### PR TITLE
Initial release.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: src/cat24c32.toit [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  /**
+  Reads $size bytes starting at $address.
+  The address must satisfy 0 <= $address <= $SIZE - $size.
+  For $size == 0, the address must also satisfy $address < $SIZE.
+  */
+  read address/int --size=1 -> ByteArray:
+    if size < 0: throw "INVALID_ARGUMENT"
+    if not 0 <= address < 4096: throw "OUT_OF_RANGE"
+    if size == 0: return #[]
+    if not address + size <= 4096: throw "OUT_OF_RANGE"
```
Comment (erikcorry) [inline src/cat24c32.toit]:
```suggestion
    if size == 0: return #[]
    if not 0 <= address <= address + size <= 4096: throw "OUT_OF_RANGE"
```

File: src/cat24c32.toit [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+  /**
+  Writes the given $bytes starting at $address.
+  The address must satisfy 0 <= $address <= $SIZE - ($bytes).size.
+  If the given $bytes is empty, then the address must also satisfy $address < $SIZE.
+  */
+  write address/int bytes/ByteArray -> none:
+    if not 0 <= address < 4096: throw "OUT_OF_RANGE"
+    if bytes.size == 0: return
+    if not address + bytes.size <= 4096: throw "OUT_OF_RANGE"
```
Comment (erikcorry) [inline src/cat24c32.toit]:
```suggestion
    if bytes.size == 0: return
    if not 0 <= address <= address + size <= 4096: throw "OUT_OF_RANGE"
```

**Review by erikcorry — APPROVED** [review]

File: src/cat24c32.toit (hunk 1 above)
Comment (floitsch) [reply src/cat24c32.toit]: done. (but in reverse order).

File: src/cat24c32.toit (hunk 2 above)
Comment (floitsch) [reply src/cat24c32.toit]: done (in reverse order).

**Pull request merged**